lib/scheduler.js

```diff
@@ -29,7 +29,7 @@ function Scheduler (options) {
     var self = this;
 
     self.options = {};
-    self.options.frameworkName = options.frameworkName.replace(/ /g, "-") || "mesos-framework." + uuid.v4();
+    self.options.frameworkName = (options.frameworkName || "mesos-framework." + uuid.v4()).replace(/ /g, "-");
```

Comment:
Fails as in the constructor crashes...
It's either this or change it to an actual if sentence, do you think that's better?